src/ert/ensemble_evaluator/snapshot.py

```diff
@@ -240,179 +208,209 @@ def _merge(self, other: "PartialSnapshot") -> "PartialSnapshot":
+                            "status": state.FORWARD_MODEL_STATE_FAILURE,
+                            "end_time": end_time,  # type: ignore
+                            "error": "The run is cancelled due to "
+                            "reaching MAX_RUNTIME",
+                        }
+                    )
+        update_event_snapshot._realization_states = self._realization_states
+        return update_event_snapshot
+
+    def _handle_forward_model_event(
+        self, event: CloudEvent, update_event_snapshot: "Snapshot"
+    ) -> "Snapshot":
+        e_source = event["source"]
+        e_type = event["type"]
+        timestamp = event["time"]
+        status = _FM_TYPE_EVENT_TO_STATUS[e_type]
+        start_time = None
+        end_time = None
+        error = None
+        if e_type == ids.EVTYPE_FORWARD_MODEL_START:
+            start_time = convert_iso8601_to_datetime(timestamp)
+        elif e_type in {
+            ids.EVTYPE_FORWARD_MODEL_SUCCESS,
+            ids.EVTYPE_FORWARD_MODEL_FAILURE,
+        }:
+            end_time = convert_iso8601_to_datetime(timestamp)
+            # Make sure error msg from previous failed run is replaced
+            error = ""
+            if event.data is not None:
+                error = event.data.get(ids.ERROR_MSG)
+
+        fm_dict = {
+            ids.STATUS: status,
+            ids.START_TIME: start_time,
+            ids.END_TIME: end_time,
+            ids.INDEX: _get_forward_model_index(e_source),
+            ids.ERROR: error,
+        }
+        if e_type == ids.EVTYPE_FORWARD_MODEL_RUNNING:
+            fm_dict[ids.CURRENT_MEMORY_USAGE] = event.data.get(ids.CURRENT_MEMORY_USAGE)
+            fm_dict[ids.MAX_MEMORY_USAGE] = event.data.get(ids.MAX_MEMORY_USAGE)
+        if e_type == ids.EVTYPE_FORWARD_MODEL_START:
+            fm_dict[ids.STDOUT] = event.data.get(ids.STDOUT)
+            fm_dict[ids.STDERR] = event.data.get(ids.STDERR)
+        self.update_forward_model(
+            _get_real_id(e_source),
+            _get_forward_model_id(e_source),
+            ForwardModel(**fm_dict),
+        )
+        update_event_snapshot._forward_model_states = self._forward_model_states
+        return update_event_snapshot
 
-    def merge_event(self, event: PartialSnapshot) -> None:
-        self._my_partial._merge(event)
+    def merge_event(self, event: "Snapshot") -> None:
+        self._merge(event)
 
     def merge(self, update_as_nested_dict: Mapping[str, Any]) -> None:
-        self._my_partial._merge(_from_nested_dict(update_as_nested_dict))
+        self._merge(Snapshot.from_nested_dict(update_as_nested_dict))
 
     def merge_metadata(self, metadata: SnapshotMetadata) -> None:
-        self._my_partial._metadata.update(metadata)
-
-    def to_dict(self) -> Dict[str, Any]:
-        return self._my_partial.to_dict()
+        """only used in gui snapshot model, which only cares about the
+        snapshot's metadata"""
+        self._metadata.update(metadata)
```

Comment:
Not returning anything?

Reply:
if this was something forgotten, was it caught by tests?

Reply:
We do not use the returned value anymore